react/src/components/ResourcePresetList.tsx

```diff
@@ -0,0 +1,239 @@
+import { filterNonNullItems, iSizeToSize, localeCompare } from '../helper';
+import { useUpdatableState } from '../hooks';
+import Flex from './Flex';
+import ResourceNumber from './ResourceNumber';
+import ResourcePresetSettingModal from './ResourcePresetSettingModal';
+import { ResourcePresetListDeleteMutation } from './__generated__/ResourcePresetListDeleteMutation.graphql';
+import { ResourcePresetListQuery } from './__generated__/ResourcePresetListQuery.graphql';
+import { ResourcePresetSettingModalFragment$key } from './__generated__/ResourcePresetSettingModalFragment.graphql';
+import {
+  ReloadOutlined,
+  PlusOutlined,
+  SettingOutlined,
+  DeleteOutlined,
+} from '@ant-design/icons';
+import { Tooltip, Button, theme, Table, App, Typography } from 'antd';
+import graphql from 'babel-plugin-relay/macro';
+import _ from 'lodash';
+import React, { Suspense, useState, useTransition } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLazyLoadQuery, useMutation } from 'react-relay';
+
+interface ResourcePresetListProps {}
```

Comment:
Why is this interface needed?

Reply:
I left it empty for later use when I want to add props. This is also because components are generated automatically, and there are quite a few empty props in other components as well ;-)